ui/app/AppLayouts/Chat/ContactsColumn/ChannelList.qml

```diff
@@ -48,6 +48,65 @@ ScrollView {
         }
     }
 
+    PopupMenu {
+        property int channelIndex
+
+        id: channelContextMenu
+        width: 175
+
+        function openMenu(channelIndex) {
+            channelContextMenu.channelIndex = channelIndex
+            channelContextMenu.popup()
+        }
+
+        Action {
+            text: qsTr("View Group")
+            icon.source: "../../../img/group.svg"
+            icon.width: 13
+            icon.height: 13
+            onTriggered: console.log('TODO View group')
+        }
+        Action {
+            text: qsTr("Mute Chat")
+            icon.source: "../../../img/bell.svg"
+            icon.width: 13
+            icon.height: 13
+            onTriggered: console.log('TODO Mute')
+        }
+        Action {
+            text: qsTr("Mark as Read")
+            icon.source: "../../../img/check-circle.svg"
+            icon.width: 13
+            icon.height: 13
+            onTriggered: {
+                chatsModel.markAllChannelMessagesReadByIndex(channelContextMenu.channelIndex)
+            }
+        }
+        Action {
+            text: qsTr("Fetch Messages")
+            icon.source: "../../../img/fetch.svg"
+            icon.width: 13
+            icon.height: 13
+            onTriggered: {
+                chatsModel.loadMoreMessagesWithIndex(channelContextMenu.channelIndex)
```

Comment:
This should display a submenu containing the options: Last 24 hours, Last  2 days, Last 3 Days, Last 7 Days